src/Compilers/VisualBasic/Portable/VBResources.resx

```diff
@@ -5695,4 +5695,16 @@
   <data name="ERR_InvalidExperimentalDiagID" xml:space="preserve">
     <value>The diagnosticId argument to the 'Experimental' attribute must be a valid identifier</value>
   </data>
+  <data name="WRN_LockTypeUnsupported" xml:space="preserve">
+    <value>A value of type 'System.Threading.Lock' in SyncLock will use likely unintended monitor-based locking. Consider manually calling 'Enter' and 'Exit' methods in a Try/Finally block instead.</value>
```

Comment:
>'Enter' and 'Exit' [](http://example.com/codeflow?start=139&length=18)

I do not see these APIs in the feature speclet. Where are they defined?

Reply:
In the runtime API proposal: https://github.com/dotnet/runtime/issues/34812

Reply:
> In the runtime API proposal: https://github.com/dotnet/runtime/issues/34812

I suggest updating the C# speclet accordingly then. Also, according to that link, it looks like the ```EnterLockScope``` is obsolete/changed, but we are still using old name.

Reply:
> I suggest updating the C# speclet accordingly then

Not sure what you mean, the C# feature doesn't use these Enter/Exit methods, so they probably shouldn't be part of the speclet?

> Also, according to that link, it looks like the `EnterLockScope` is obsolete/changed, but we are still using old name.

Yes, I know, waiting for the API to be approved and then I will change to the final name.

Reply:
> Not sure what you mean, the C# feature doesn't use these Enter/Exit methods, so they probably shouldn't be part of the speclet?

Aren't these APIs part of the type. I do not see any harm in mentioning them. But I do see a benefit. At least we wouldn't have this conversation if the speclet had the information.

Reply:
Just adding them to the Lock type as mentioned in the speclet today would probably imply we check for existence of those methods and fail if they are missing. But I guess I can mention them separately. https://github.com/dotnet/csharplang/pull/7921

Reply:
> Just adding them to the Lock type as mentioned in the speclet today would probably imply we check for existence of those methods and fail if they are missing.

No would not. Definition of the type is just a definition of the type. There is no requirement to drop un-interesting members. The only motivation is to save some space in the document. But these members are interesting. Presence of which members and under which conditions is important for the feature is explicitly stated elsewhere in the speclet.